_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,185 @@
+You might already be familiar to this task from upgrading Foreman to a newer
+version.
+
+Command `foreman-rake -T` lists all available tasks. We will be using
+`foreman-rake db:convert:prod2dev` to migrate the data.
+
+Foreman internally uses a development framework called Rails. In Rails, you can
+have a specific configuration for e.g. a development and a production
+environment. Our prod2dev rake task assumes, that the source database is
+configured as the production database and the new target database is configured
+as the development database. All data in the development database will be
+overwritten.
+
+You can configure the new database as Foreman's new development database by
+adding a new connection in `/etc/foreman/database.yml`.
+
+It may look like this:
+
+```yaml
+# The new target database
+development:
+  adapter: postgresql
+  host: postgres.example.com
+  port: 5432
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+
+# The old source database
+production:
+  adapter: mysql2
+  host: mysql.example.com
+  port: 3306
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+```
+
+The new database needs to be set up with all the tables required for Foreman to
+operate, so in order to prepare the new database to receive the actual data, we
+need to run all of Foreman's database migrations.
+
+```sh
+forman-rake db:migrate RAILS_ENV=development
+```
+
+Next, we can finally migrate all the data. This will take a long time depending
+on the size of your database. It is highly recommended to run this in a screen
+or tmux session.
+
+```sh
+forman-rake db:convert:prod2dev
+
+```
+
+#### Wrapping up
+
+Postgresql uses sequences in order to store the next free primary key id for a
+table. These sequences do not get updated by the rake tasks and have to be
+adjusted manually. On MySQL, this is not needed as the concept of sequences is
+not known there.
+
+The Postgresql Wiki has a [script to repair the
+sequences](https://wiki.postgresql.org/wiki/Fixing_Sequences). Copy the script
+and save it to a file, e.g. `sequences.sql`. Then run it like this:
+
+```sh
+su - postgres
+vim sequences.sql
+# Paste the script and save the file
+psql foreman -Atq -f reset.sql -o temp
+psql -f temp --username postgres foreman
+rm temp
+```
+
+Then, make sure to resore the original permissions of your foreman database
+user. This can be done on Postgresql by issuing this SQL command.
+
+```sql
+ALTER ROLE "foreman" NOSUPERUSER;
+```
+
+To switch your Foreman instance to the new database, you need to change the
+production database configuration in `/etc/foreman/database.yml` to point to
+the development installation we configured earlier. You can either do this
+manually or re-run foreman-installer with the new parameters for Postgresql.
+
+Then you can finally start Foreman's services again.
+
+```sh
+systemctl start httpd.service
+systemctl start foreman-tasks.service
```

Comment:
doesn't httpd also start tasks now?

Reply:
I'm not aware of it. Why would it do that?

Reply:
No, it doesn't the installer ensures it's running. The current name of `foreman-tasks` is `dynflowd` btw. `foreman-tasks` is a legacy alias.

Reply:
Ok, ignore me, i was thinking of the dev environment which also starts dynflow.

Reply:
sould still be good to change the service name though

Reply:
Changed to dynflowd then.